### PR TITLE
feat: variadic input helpers

### PR DESCRIPTION
**Pull request by didavid61202**
## Update & Proposal
Update input helpers to take variadic arguments, eg: `exactly('foo', maybe('bar'), 'baz'...)`, enabling a more concise syntax and improving the overall readability of the code.

Currently, the input helpers take only one argument of either type `string` or another `Input`. To concatenate different RegExp patterns, users need to use the `.and()` chaining helper repeatedly. This approach is very descriptive but can also make the code verbose and difficult to read when dealing with multiple input helpers in more complex scenario.

To improve the syntax and readability, I want to propose an update to all input helpers to variadic function that take one or more `string` or `Input` as arguments.
And this update will maintain backward compatibility, as the new helpers still accept a single argument of type `string` or `Input`, and still can chain using `.and()`. However, users will also have the option to pass multiple arguments to the functions, which will be concatenated accordingly.

please let me know your suggestion, feedback, and input.

## Updated inputs
- [x] `createRegExp` (last array arg as flags)
- [x] `maybe`
- [x] `exactly`
- [x] `oneOrMore`
- [x] `.and`
- [x] `.or`
- [x] `.after`
- [x] `.before`
- [x] `.notAfter`
- [x] `.notBefore`

## Example
original `semver` example
```ts
const regExp = createRegExp(
  oneOrMore(digit)
    .groupedAs('major')
    .and('.')
    .and(oneOrMore(digit).groupedAs('minor'))
    .and(exactly('.').and(oneOrMore(char).groupedAs('patch')).optionally())
)
```
with this update, can be simplified to:
```ts
  const regExp2 = createRegExp(
      oneOrMore(digit).groupedAs('major'),
      '.',
      oneOrMore(digit).groupedAs('minor'),
      maybe('.', oneOrMore(char).groupedAs('patch'))
    )
```

## Related issues
#237 

 - [x] add/update tests
 - [x] update docs
 - [x] update jsdoc comments

### CODE REVIEW

**Comment by danielroe** [comment]
I really like this ❤️  It makes writing regexp much less verbose - great DX.

**Comment by didavid61202** [comment]
> I really like this ❤️ It makes writing regexp much less verbose - great DX.

@danielroe  Thanks! 
Always strive for better DX 🚀 
Just complete updating the docs and JSDoc comment to all related input helpers too, please check on them if you got some spare time, then I think it's good to go 👍

**Comment by xRSquared** [comment]
Using variadic functions is an elegant solution. I love it!

File: docs/content/2.getting-started/2.usage.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ console.log(regExp)
 
 Every pattern you create with the library should be wrapped in `createRegExp`, which enables the build-time transform.
 
-The first argument is either a string to match exactly, or an input pattern built up using helpers from `magic-regexp`. It also takes a second argument, which is an array of flags or flags string.
+`createRegExp` accepts arbitrary number of arguments of type `string`, or `Input` that are built up using helpers from `magic-regexp`, and an optional final argument of an array of flags or flags string. It creates a `MagicRegExp`, which has all the patterns from the inputs concatenated.
```
Comment (danielroe) [inline docs/content/2.getting-started/2.usage.md:18]:
```suggestion
`createRegExp` accepts an arbitrary number of arguments of type `string` or `Input` (built up using helpers from `magic-regexp`), and an optional final argument of an array of flags or a flags string. It creates a `MagicRegExp`, which concatenates all the patterns from the arguments that were passed in.
```

File: docs/content/2.getting-started/2.usage.md [hunk 2]
```diff
@@ -25,6 +25,16 @@ createRegExp(exactly('foo').or('bar'))
...
+
+// or pass in multiple `string` and `input patterns`,
+// all inputs will be concatenated to one RegExp pattern
+createRegExp(
+    'foo',
+    maybe('bar').groupedAs('g1'),
+    'baz',
+    [global, multiline]
+) 
+// equivalent to /foo(?<g1>(?:bar)?)baz/gm
```
Comment (danielroe) [inline docs/content/2.getting-started/2.usage.md:37]:
```suggestion
// or pass in multiple `string` and `input patterns`,
// all inputs will be concatenated to one RegExp pattern
createRegExp(
  'foo',
  maybe('bar').groupedAs('g1'),
  'baz',
  [global, multiline]
) 
// equivalent to /foo(?<g1>(?:bar)?)baz/gm
```

**Pull request merged**